### PR TITLE
Update busybox image link to gcr.io for kube-proxy

### PR DESCRIPTION
**Pull request by MrHohn**
**What this PR does / why we need it**: Point busybox image to gcr.io.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #NONE

**Special notes for your reviewer**:
@ixdy @bowei 

**Release note**:

```release-note
Update busybox image link to gcr.io for kube-proxy.
```

### CODE REVIEW

File: cluster/saltbase/salt/kube-proxy/kube-proxy.manifest [hunk 1]
```diff
@@ -56,7 +56,7 @@ spec:
   hostNetwork: true
   initContainers:
   - name: touch-lock
-    image: busybox
```
Comment (ixdy) [inline cluster/saltbase/salt/kube-proxy/kube-proxy.manifest]: `gcr.io/google_containers/busybox:1.24` is pretty old (much older than `:latest` in dockerhub) - why are we using it?

**Comment by cblecker** [comment]
/hold
for image update per https://github.com/kubernetes/kubernetes/pull/53819#discussion_r144384731

File: cluster/saltbase/salt/kube-proxy/kube-proxy.manifest (hunk 1 above)
Comment (MrHohn) [reply cluster/saltbase/salt/kube-proxy/kube-proxy.manifest]: Updated to `gcr.io/google_containers/busybox:1.27`, corresponding to `busybox:1.27`.

**Comment by cblecker** [comment]
/hold cancel
:)

**Comment by bowei** [comment]
You need to get cherry pick approved.

**Comment by MrHohn** [comment]
/assign @wojtek-t 
for cherrypick approval.

**Comment by wojtek-t** [comment]
I'm obviously fine with cherrypicking.
Can you please create a cherrypick and add a release note?

**Comment by MrHohn** [comment]
@wojtek-t Thanks! Release note added to this PR.

**Pull request merged**